Realm/RLMObjectSchema.mm

```diff
@@ -118,7 +118,9 @@ + (instancetype)schemaForObjectClass:(Class)objectClass {
     if (NSString *primaryKey = [objectClass primaryKey]) {
         for (RLMProperty *prop in schema.properties) {
             if ([primaryKey isEqualToString:prop.name]) {
-                prop.indexed = YES;
+                if (prop.type == RLMPropertyTypeString) {
```

Comment:
This is correct. Int indexing does not work. It was reverted on master. This reflects those changes on this branch